src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -307,13 +326,31 @@ public PointSetTimes eval (TravelTimeFunction travelTimeForVertex, int offstreet
             int time0 = travelTimeForVertex.getTravelTime(edge.getFromVertex());
             int time1 = travelTimeForVertex.getTravelTime(edge.getToVertex());
 
-            // TODO apply walk speed
+            // An "off-roading" penalty is applied to limit extending isochrones into water bodies, etc.
+            // We may want to keep the MAX_OFFSTREET_WALK_METERS relatively high to avoid holes in the isochrones,
+            // but make it costly to walk long distances where there aren't streets.  The approach below
+            // accomplishes that, applying a penalty to off-street distances greater than the typical grid cell size.
+            // We could use a distance threshold more closely tied to pointset resolution/coverage
+            int scaledLinkDist = distancesToEdge_mm[i] > CELL_SIZE*1000 ?
+                    distancesToEdge_mm[i] + OFFROAD_PENALTY * (distancesToEdge_mm[i] - CELL_SIZE * 1000) : distancesToEdge_mm[i];
```

Comment:
I'm hesitant to use empirically tuned unitless multipliers to keep the isochrone edges from extending out into the water. They would probably work in one place and not another, and seem to be prone to overflow. Also, because they are pre-multiplied into the distance we lose information that could be useful for dynamically applying speed or distance parameters during routing. It's also hard to justify why speed would change at some predetermined distance - if we consider a cell within reachable distance at all, how do we know where the hypothetical walking path to that cell ends? Instead I think we should just limit the linking distance, then apply the same walking/biking speed across the full length.